### PR TITLE
web support

### PR DESCRIPTION
**Pull request by EvanBacon**
# Summary

Uses the CSS API to observe the safe-area-inset on web.

## Test Plan

[Example video](https://photos.app.goo.gl/BHMbhb6FNMVRiV8f6)

### What's required for testing (prerequisites)?

Phone with notch, i.e: iPhone X

### What are the steps to reproduce (after prerequisites)?

Run `expo start --web` in the `/example`. Visit the page from a device with a notch, then turn the device to landscape orientation, the black box should respect safe area insets.

You can also try saving the page as a PWA to get the full effect.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |
| Web |    ✅     |

## Checklist

- [x] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typed files (TS and Flow)
- [x] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

File: SafeView.web.js [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+    elStyle[key] = val;
+  }
+}
+
+const passiveEvents = getPassiveEvents();
+
+export default class SafeView extends React.Component {
+  static setStatusBarHeight = height => {
+    console.warn('SafeView.setStatusBarHeight() is not supported on web');
+    _customStatusBarHeight = height;
```
Comment (brentvatne) [inline SafeView.web.js]: probably just remove this line

File: SafeView.web.js [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+    const a2Children = document.createElement('div');
+    const W = 100;
+    const MAX = 10000;
+    const aStyle = {
+      position: 'absolute',
+      width: `${W}px`,
+      height: '200px',
+      boxSizing: 'border-box',
+      overflow: 'hidden',
+      paddingBottom: `${support}(safe-area-inset-${attr})`,
```
Comment (necolas) [inline SafeView.web.js]: You can end up coercing `null` here.

File: SafeView.web.js [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+  };
+
+  for (const t in transitions) {
+    if (el.style[t] !== undefined) {
+      return transitions[t];
+    }
+  }
+}
+
+function getSupportedEnv() {
```
Comment (necolas) [inline SafeView.web.js]: This check could probably be as simple as `env` by default and `constant` is supported

**Review by necolas — COMMENTED** [review]
The web implementation of this component is so complicated and heavy that it would be a concern to see this being used in a web app. Why is `Animated` needed, for example?

**Comment by opeologist** [comment]
is there any update on this PR / web support for this package?  i'm seeing an issue coming from react-native-safe-area-view when attempting to use react-navigation-stack on web with the head of react-native-gesture-handler that just added web support a few days ago (thanks again @EvanBacon ) that looks like the web version of this package isn't exporting the getStatusBarHeight function, which is used in react-navigation-stack's getDefaultHeaderHeight function.  not sure if this PR resolves that issue, and since it seems a bit stale, i'm not sure if attempting to use the branch of this PR would only make matters worse or not.

**Comment by opeologist** [comment]
after stitching this branch into react-navigation-stack, it looks like the issue is that the function getStatusBarHeight isn't exported in the index at all, which react-navigation-stack is spreading to gain access to (https://github.com/react-navigation/stack/blob/master/src/views/Header/HeaderSegment.tsx#L10)

**Comment by EvanBacon** [comment]
@necolas IIRC this is so you can set style changes without updating the tree like: `this.setNativeProps({ style: { display: "none" } });`

**Comment by EvanBacon** [comment]
@tehOPEologist this PR isn't implementing `forceInset` yet. It'd probably require that some tests are written before I felt comfortable merging this.

**Comment by vitalyiegorov** [comment]
@EvanBacon Do you have any updates? Or maybe you can advise any workaround?

**Comment by brentvatne** [comment]
it seems like this should be a lot easier now that you've added the logic for it to https://github.com/th3rdwave/react-native-safe-area-context! :O

**Comment by brentvatne** [comment]
react-native-safe-area-context supports web and this lib is deprecated so yeet

**Comment by EvanBacon** [comment]
🥸🥸🥸

**Pull request closed without merging**